test/PrintAsObjC/cdecl-official.swift

```diff
@@ -0,0 +1,85 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t --leading-lines
+
+/// Generate cdecl.h
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) \
+// RUN:   %t/Lib.swift -emit-module -verify -o %t -emit-module-doc \
+// RUN:   -emit-objc-header-path %t/cdecl.h \
+// RUN:   -enable-experimental-feature CDecl
+
+/// Check cdecl.h directly
+// RUN: %FileCheck %s --input-file %t/cdecl.h
+// RUN: %check-in-clang %t/cdecl.h
+// RUN: %check-in-clang-c %t/cdecl.h -Wnullable-to-nonnull-conversion
+// RUN: %check-in-clang-cxx %t/cdecl.h
+
+/// Build a client against cdecl.h
+// RUN: %clang -c %t/Client.c -fmodules -I %t \
+// RUN:   -F %S/../Inputs/clang-importer-sdk-path/frameworks \
+// RUN:   -I %clang-include-dir -Werror \
+// RUN:   -isysroot %S/../Inputs/clang-importer-sdk
+
+// REQUIRES: swift_feature_CDecl
+
+//--- Lib.swift
+
+// CHECK: #if defined(__cplusplus)
+// CHECK: extern "C" {
+// CHECK: #endif
+
+/// My documentation
+@cdecl("simple")
+func a_simple(x: Int, bar y: Int) -> Int { return x }
+// CHECK-LABEL: // My documentation
+// CHECK-LABEL: SWIFT_EXTERN ptrdiff_t simple(ptrdiff_t x, ptrdiff_t y) SWIFT_NOEXCEPT SWIFT_WARN_UNUSED_RESULT;
+
+@cdecl("primitiveTypes")
+public func b_primitiveTypes(i: Int, ci: CInt, l: CLong, c: CChar, f: Float, d: Double, b: Bool) {}
+// CHECK-LABEL: SWIFT_EXTERN void primitiveTypes(ptrdiff_t i, int ci, long l, char c, float f, double d, bool b) SWIFT_NOEXCEPT;
+
+@cdecl("has_keyword_arg_names")
+func c_keywordArgNames(auto: Int, union: Int) {}
+// CHECK-LABEL: SWIFT_EXTERN void has_keyword_arg_names(ptrdiff_t auto_, ptrdiff_t union_) SWIFT_NOEXCEPT;
```

Comment:
I would add test cases for various pointer types: `UnsafeMutableRawPointer` should become `void * _Nonnull`, `UnsafeMutablePointer<Int>` should become `ptrdiff_t * _Nonnull`, the non-`Mutable` variants should gain a `const`, and adding `?` or `!` at the end should get you `_Nullable` and `_Null_unspecified` respectively.

Reply:
Good point, I'm taking note of these. I plan on adding more test and type-checking around the types accepted by `@cdecl` in a future PR, a few more are not covered either.

Reply:
We already get the desired behavior so I've added tests for pointer types and nullability to this PR.